File: leetcode/duplicates.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  if (nums !== []){
    var countObj = nums.reduce(function (acc, curr) {
    if (typeof acc[curr] == 'undefined') {
      acc[curr] = 1;
    } else {
      acc[curr] += 1;
    }
  
    return acc;
}, {});
    for (num in countObj) {
      if (countObj[num] > 1) {
        return true;
      }
    }
  }
  return false;
};

// Runtime: 104 ms, faster than 26.51% of JavaScript online submissions for Contains Duplicate.
// Memory Usage: 47.9 MB, less than 5.18% of JavaScript online submissions for Contains Duplicat

File: leetcode/duplicates.rb
# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
  new_nums = nums.each_with_object(Hash.new(0)) do |element, num_hash|
    num_hash[element] += 1
  end 
   new_nums.values.any? {|num| num > 1} ? true : false
end

# Runtime: 84 ms, faster than 11.00% of Ruby online submissions for Contains Duplicate.
# Memory Usage: 214.4 MB, less than 6.70% of Ruby online submissions for Contains Duplicate.